src/turnkeyml/build/ignition.py

```diff
@@ -100,10 +34,46 @@ def validate_cached_model(
     the cached model is valid to use in the build.
     """
 
-    result = []
+    # The first few cache miss conditions (e.g., failed build, failed state load,
+    # out-of-date package) return that specific condition as the reason the cache
+    # missed. Any other conditions are aggregated into a multi-condition return
+    # message.
+
+    # We required that a build was successful to load it from cache
+    if vars(cached_state).get(fs.Keys.BUILD_STATUS) != build.FunctionStatus.SUCCESSFUL:
+        return ["Your cached build was not successful."]
+
+    # All of the state properties analyzed by this function
+    # should be listed here.
+    cache_analysis_properties = [
+        fs.Keys.BUILD_NAME,
+        fs.Keys.ONNX_OPSET,
+        fs.Keys.DEVICE,
+        fs.Keys.SEQUENCE,
+        fs.Keys.BUILD_STATUS,
+        fs.Keys.TURNKEY_VERSION,
+        fs.Keys.MODEL_HASH,
+        fs.Keys.EXPECTED_INPUT_DTYPES,
+        fs.Keys.EXPECTED_INPUT_SHAPES,
+        fs.Keys.DOWNCAST_APPLIED,
+        fs.Keys.UID,
+        fs.Keys.EVALUATION_ID,
+        fs.Keys.MODEL_TYPE,
+    ]
+
+    # Make sure the cached state contains all information needed to assess a cache hit
+    # so that we don't hit an attribute error down the function
+    for key in cache_analysis_properties:
+        if key not in vars(cached_state).keys():
+            return [
+                (
+                    f"Your cached build is missing state key {key}. This is a bug in "
```

Comment:
Should we advice the user to clear the cache here?
Can we consider this a cache miss and move on?

Reply:
This is considered a cache miss and the model will be automatically rebuilt - that's what happens with all the messages returned from this function.